OpenSSL-Strategy.md

```diff
@@ -0,0 +1,94 @@
+# Node.js OpenSSL Strategy
+
+The Node.js source tree contains a copy of OpenSSL. By default, Node.js compiles its own copy of OpenSSL and statically links this library into its binary. The `configure` tool allows for dynamic linking against a shared OpenSSL library. Using a combination of the `--shared-openssl`, `--shared-openssl-includes`, `--shared-openssl-libname` and `--shared-openssl-libpath` arguments to `configure`, compilation of Node.js will skip over its bundled copy of the OpenSSL source and link to dynamic library version of OpenSSL, or a suitably compatible OpenSSL equivalent. Whether this linking is possible is left for the compiler to determine at build and linking time.
```

Comment:
Nits/suggestions:

```diff
-By default, Node.js compiles its own copy of OpenSSL and statically links this library into its binary.
+By default, the Node.js build process compiles its own copy of OpenSSL and statically links this library into the built `node` binary.
```

```diff
-link to dynamic library version of OpenSSL
+link to a dynamic library version of OpenSSL
```

```diff
-at build and link time.
+at build and linking time.
```

Also it would be good to format to 80 columns, not least so leaving PR comments is easier.

Reply:
resolved